stonesoup/models/transition/orbital/orbit.py

```diff
@@ -370,6 +371,205 @@ def transition(self, orbital_state, time_interval=timedelta(seconds=0)):
+        float, default=1e-8, doc="The stopping point in the progression of "
+                                 "ever smaller f/f' ratio in the the Newton's "
+                                 "method calculation of universal anomaly."
+    )
+
+    @property
+    def ndim_state(self):
+        """Dimension of the state vector is 6
+
+        Returns
+        -------
+        : int
+            The dimension of the state vector, i.e. 6
+
+        """
+        return 6
+
+    def covar(self, time_interval=timedelta(seconds=1)):
+        r"""Return the transition covariance matrix
+
+        Parameters
+        ----------
+        time_interval: :math:`\delta t` :attr:`datetime.timedelta`, optional
+            The time interval over which to test the new state (default is 1
+            second)
+
+        Returns
+        -------
+        : CovarianceMatrix
+            The transition covariance matrix
+
+        """
+        return CovarianceMatrix(self.noise * time_interval.total_seconds())
+
+    def function(self, orbital_state, noise=0,
+                 time_interval=timedelta(seconds=0)):
+        r"""Just passes parameters to the transition function
+
+        Parameters
+        ----------
+        orbital_state : :class:`~.OrbitalState`
+            The prior orbital state
+        noise : CovarianceMatrix, optional
+            The nominal covariance matrix. This isn't passed to the transition
+            function though, so can be left out.
+        time_interval: :math:`\delta t` :attr:`datetime.timedelta`, optional
+            The time interval over which to test the new state (default is 0
+            seconds)
+
+        Returns
+        -------
+        : StateVector
+            The orbital state vector returned by the transition function
+
+        Note
+        ----
+        This merely passes the parameters to the :attr:`.transition()`
+        function. The noise parameter has no effect, is included merely
+        for compatibility with parent classes.
+        """
+        return self.transition(orbital_state, time_interval)
+
+    def transition(self, orbital_state, time_interval=timedelta(seconds=0)):
+        r"""The transition proceeds as algorithm 3.4 in [1]
+
+        Parameters
+        ----------
+        orbital_state : :class:`~.OrbitalState`
+            The prior orbital state
+        time_interval: :math:`\delta t` :attr:`datetime.timedelta`, optional
+            The time interval over which to test the new state (default
+            is 0 seconds)
+
+        Returns
+        -------
+        : StateVector
+            The orbital state vector returned by the transition function
+
+        Warning
+        -------
+        If noisy samples from the transition function are required, use
+        the :attr:`rvs` method.
+        """
+        # Get the position and velocity vectors
+        # Evaluated at initial timestamp
+        tle_ext = Satrec.twoline2rv(orbital_state.state.metadata['line_1'],
```

Comment:
This works because I have input a ground truth state. I'm unsure if the transition model should or should not be able to work with GroundTruthState object. I could always extract the Orbital State prior to entering the transition function.